### PR TITLE
expose parseSource

### PR DESCRIPTION
**Pull request by woodgear**

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Can you please explain the change with words?

**Comment by woodgear** [comment]
expose a mehod named parseSource，that accept the sourcecode string as argument，since that we could use this api，the parse our code more programly。

File: lib/index.js [hunk 1]
```diff
@@ -33,6 +33,15 @@ var defaults = {
...
 
+// Simple logger interace
+var logger = {
+    debug  : function() { console.log(arguments); },
+    verbose: function() { console.log(arguments); },
+    info   : function() { console.log(arguments); },
+    warn   : function() { console.log(arguments); },
+    error  : function() { console.log(arguments); }
+};
+
```
Comment (tommy87) [inline lib/index.js:44]: Why you moved the logger above. I assume it has something todo with issue #105. But the you should create an own merge request for this.

File: lib/index.js [hunk 2]
```diff
@@ -299,6 +299,54 @@ function parse(options) {
...
+        new PluginLoader(app);
+        var parser = new Parser(app);
+        var worker = new Worker(app);
+        var filter = new Filter(app);
+
+        // Make them available for plugins
+        app.parser = parser;
+        app.worker = worker;
+        app.filter = filter;
```
Comment (tommy87) [inline lib/index.js:342]: Until here your whole code is the same as in `function parse(options) { ` So you should put this parts in an extra function or combine both functions to one. Doubling code is never a good idea

**Review by tommy87 — CHANGES_REQUESTED** [review]
Please also write a test in `apidoc-core/test/` that show that loading from file and loading from source are working at the same time

**Comment by woodgear** [comment]
@tommy87 @NicolasCARPi  i have updated this pr by review.

**Review by tommy87 — APPROVED** [review]
Looks good for me, but the code should be tested before merging.

**Comment by woodgear** [comment]
i glad to add some test either,but i thought i need some help for that. for example where should i add the test to? [here](https://github.com/apidoc/apidoc-core/blob/8ead7e7ed9faa13c4f870a82ab22e0d17bc76561/test/apidoc_test.js#L100)?

**Comment by tommy87** [comment]
Just create a new file [here](https://github.com/apidoc/apidoc-core/blob/8ead7e7ed9faa13c4f870a82ab22e0d17bc76561/test/parser_api_description_test.js) is good and short example how you could do that

**Comment by woodgear** [comment]
@tommy87  @NicolasCARPi  add some test for parseSource

**Comment by tommy87** [comment]
@woodgear i have add some CI action to the dev branch. Please rebase your code, so that the tests are triggered. Oh and please merge into the dev branch, not in the master branch.

**Comment by woodgear** [comment]
created a new pr in #109

**Pull request closed without merging**